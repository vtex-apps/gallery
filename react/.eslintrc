{
  "extends": "vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  },
  "rules": {
    "react/jsx-indent": "off"
  }
}
