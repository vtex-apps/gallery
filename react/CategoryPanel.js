import React, { Fragment } from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { Link } from 'vtex.render-runtime'
import { facetOptionShape } from './constants/propTypes'
import searchResult from './searchResult.css'
import CategoriesHighlights from 'vtex.store-components/CategoriesHighlights'

const renderCategoryShelf = (category, noOfChildren, quantityOfItemsPerRow) => {
  const categoriesHighlighted = category.children
    .slice(0, noOfChildren)
    .map(child => ({
      id: child.id,
      name: child.name,
      to: child.link,
    }))
  const headerClasses = classNames(
    searchResult.categoryPanelHeaderRow,
    'flex flex-row flex-wrap items-stretch pa3 bn mh4-ns'
  )
  const itemClasses = classNames(
    searchResult.categoryPanelItemRow,
    'flex flex-row flex-wrap items-stretch pa3 bn mh4-ns'
  )
  let i = 0
  const items = []
  for (i = 0; i < categoriesHighlighted.length; i += quantityOfItemsPerRow) {
    items.push(categoriesHighlighted.slice(i, i + quantityOfItemsPerRow))
  }
  return (
    <Fragment key={`parent-fragment-${category.id}`}>
      <h3 className={`t-heading-3 ${headerClasses}`}>
        <Link to={category.link}>{category.name}</Link>
      </h3>
      {items.length > 0 &&
        items.map((item, index) => (
          <div key={`category-row-${index}`} className={itemClasses}>
            <CategoriesHighlights
              categoriesHighlighted={item}
              showCategoriesHighlighted
              quantityOfItems={quantityOfItemsPerRow}
            />
          </div>
        ))}
      {category.children.map(child => {
        if (child.children && child.children.length > 0) {
          return renderCategoryShelf(child, noOfChildren, quantityOfItemsPerRow)
        }
      })}
    </Fragment>
  )
}

const CategoryPanel = ({ tree, noOfCategories, quantityOfItemsPerRow }) => {
  return (
    <div>
      {tree.map(category => {
        return renderCategoryShelf(
          category,
          noOfCategories,
          quantityOfItemsPerRow
        )
      })}
    </div>
  )
}

CategoryPanel.propTypes = {
  /** Category tree */
  tree: PropTypes.arrayOf(facetOptionShape),
  /** Number of category filters to be shown */
  noOfCategories: PropTypes.number,
  /** Number of categories per row */
  quantityOfItemsPerRow: PropTypes.number,
}

CategoryPanel.defaultProps = {
  tree: [],
  noOfCategories: 1000,
  quantityOfItemsPerRow: 4,
}

export default CategoryPanel
