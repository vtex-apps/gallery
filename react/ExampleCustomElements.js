import React, { Component } from 'react'
import SearchResults from './SearchResult'

const GalleryItem=(props) =><div>Im a card </div>
const Gallery=(props) =><div>Im a Gallery </div>
const SelectedFilters=(props) =><div>Im Selected Filters</div>
const SearchFilter=(props) =><div>Im a Filter </div>
const Spinner=(props) =><div>Loading!!! </div>

export default (props)=>{
    return <SearchResults {...props} 
        //GalleryItem={GalleryItem} 
        //Gallery={Gallery}
        //SearchFilter={SearchFilter}
        Spinner={Spinner}
        //SelectedFilters={SelectedFilters}
    />
}