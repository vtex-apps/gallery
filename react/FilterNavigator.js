import React from 'react'
import PropTypes from 'prop-types'
import { useRuntime } from 'vtex.render-runtime'
import { flatten, path, identity, contains } from 'ramda'
import ContentLoader from 'react-content-loader'
import classNames from 'classnames'
import { FormattedMessage } from 'react-intl'

import FilterSidebar from './components/FilterSidebar'
import SelectedFilters from './components/SelectedFilters'
import AvailableFilters from './components/AvailableFilters'
import {
  formatCategoriesTree,
  mountOptions,
  getMapByType,
} from './constants/SearchHelpers'
import {
  facetOptionShape,
  paramShape,
  hiddenFacetsSchema,
} from './constants/propTypes'

import searchResult from './searchResult.css'

export const CATEGORIES_TYPE = 'Categories'
export const BRANDS_TYPE = 'Brands'
export const PRICE_RANGES_TYPE = 'PriceRanges'
export const SPECIFICATION_FILTERS_TYPE = 'SpecificationFilters'

const CATEGORIES_TITLE = 'search.filter.title.categories'
const BRANDS_TITLE = 'search.filter.title.brands'
const PRICE_RANGES_TITLE = 'search.filter.title.price-ranges'

/**
 * Wrapper around the filters (selected and available) as well
 * as the popup filters that appear on mobile devices
 */
const FilterNavigator = ({
  query,
  map,
  priceRange,
  getLinkProps,
  tree = [],
  specificationFilters = [],
  priceRanges = [],
  brands = [],
  loading = false,
  hiddenFacets = {},
}) => {
  const {
    hints: { mobile },
  } = useRuntime()

  const getCategories = () => {
    if (!tree || !tree.length) {
      return []
    }
    return formatCategoriesTree(tree)
  }

  const getAvailableCategories = (showOnlySelected = false) => {
    let queryParams = query || ''

    const mapArray = map.split(',')

    const categoriesCount = mapArray.filter(
      m => m === getMapByType(CATEGORIES_TYPE)
    ).length

    const currentPath = queryParams
      .split('/')
      .filter((_, index) => mapArray[index] === getMapByType(CATEGORIES_TYPE))
      .join('/')

    return getCategories()
      .filter(c =>
        showOnlySelected
          ? c.level === categoriesCount - 1
          : c.level === categoriesCount - 1 || c.level === categoriesCount
      )
      .filter(c => c.path.toLowerCase().startsWith(currentPath.toLowerCase()))
  }

  const getSelectedFilters = () => {
    const availableCategories = getAvailableCategories(true)

    const options = [
      ...mountOptions(availableCategories, CATEGORIES_TYPE, map),
      ...specificationFilters.map(spec =>
        mountOptions(spec.facets, SPECIFICATION_FILTERS_TYPE, map)
      ),
      ...mountOptions(brands, BRANDS_TYPE, map),
      ...mountOptions(priceRanges, PRICE_RANGES_TYPE, map),
    ]

    return flatten(options).filter(opt => opt.selected)
  }

  const getFilters = () => {
    const categories = getAvailableCategories()

    const hiddenFacetsNames = (
      path(['specificationFilters', 'hiddenFilters'], hiddenFacets) || []
    ).map(filter => filter.name)

    const mappedSpecificationFilters = !path(
      ['specificationFilters', 'hideAll'],
      hiddenFacets
    )
      ? specificationFilters
          .filter(spec => !contains(spec.name, hiddenFacetsNames))
          .map(spec => ({
            type: SPECIFICATION_FILTERS_TYPE,
            title: spec.name,
            options: spec.facets,
          }))
      : []

    return [
      !hiddenFacets.categories && {
        type: CATEGORIES_TYPE,
        title: CATEGORIES_TITLE,
        options: categories,
        oneSelectedCollapse: true,
      },
      ...mappedSpecificationFilters,
      !hiddenFacets.brands && {
        type: BRANDS_TYPE,
        title: BRANDS_TITLE,
        options: brands,
      },
      !hiddenFacets.priceRange && {
        type: PRICE_RANGES_TYPE,
        title: PRICE_RANGES_TITLE,
        options: priceRanges,
      },
    ].filter(identity)
  }

  const filterClasses = classNames({
    'flex justify-center flex-auto ': mobile,
  })

  if (!map || !map.length) {
    return null
  }

  if (loading && !mobile) {
    return (
      <div className={searchResult.filters}>
        <ContentLoader
          style={{
            width: '100%',
            height: '100%',
          }}
          width="230"
          height="320"
          y="0"
          x="0"
        >
          <rect width="100%" height="1em" />
          <rect width="100%" height="8em" y="1.5em" />
          <rect width="100%" height="1em" y="10.5em" />
          <rect width="100%" height="8em" y="12em" />
        </ContentLoader>
      </div>
    )
  }

  if (mobile) {
    return (
      <div className={searchResult.filters}>
        <div className={filterClasses}>
          <FilterSidebar
            filters={getFilters()}
            selectedFilters={getSelectedFilters()}
            getLinkProps={getLinkProps}
            map={map}
          />
        </div>
      </div>
    )
  }

  return (
    <div className={searchResult.filters}>
      <div className={filterClasses}>
        <div className="bb b--muted-4">
          <h5 className="t-heading-5 mv5">
            <FormattedMessage id="search-result.filter-button.title" />
          </h5>
        </div>
        <SelectedFilters
          filters={getSelectedFilters()}
          getLinkProps={getLinkProps}
        />
        <AvailableFilters
          filters={getFilters()}
          map={map}
          priceRange={priceRange}
        />
      </div>
    </div>
  )
}

FilterNavigator.propTypes = {
  /** Get the props to pass to render's Link */
  getLinkProps: PropTypes.func.isRequired,
  /** Categories tree */
  tree: PropTypes.arrayOf(facetOptionShape),
  /** Params from pages */
  params: paramShape,
  /** List of brand filters (e.g. Samsung) */
  brands: PropTypes.arrayOf(facetOptionShape),
  /** List of specification filters (e.g. Android 7.0) */
  specificationFilters: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      facets: PropTypes.arrayOf(facetOptionShape),
    })
  ),
  /** List of price ranges filters (e.g. from-0-to-100) */
  priceRanges: PropTypes.arrayOf(facetOptionShape),
  /** Current price range filter query parameter */
  priceRange: PropTypes.string,
  /** Map query parameter */
  map: PropTypes.string,
  /** Loading indicator */
  loading: PropTypes.bool,
  ...hiddenFacetsSchema,
}

export default FilterNavigator
