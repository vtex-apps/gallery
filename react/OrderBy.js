import PropTypes from 'prop-types'
import React, { useMemo } from 'react'
import { injectIntl, intlShape } from 'react-intl'

import SelectionListOrderBy from './components/SelectionListOrderBy'

import searchResult from './searchResult.css'

export const SORT_OPTIONS = [
  {
    value: '',
    label: 'ordenation.sort-by',
  },
  {
    value: 'OrderByTopSaleDESC',
    label: 'ordenation.sales',
  },
  {
    value: 'OrderByReleaseDateDESC',
    label: 'ordenation.release.date',
  },
  {
    value: 'OrderByBestDiscountDESC',
    label: 'ordenation.discount',
  },
  {
    value: 'OrderByPriceDESC',
    label: 'ordenation.price.descending',
  },
  {
    value: 'OrderByPriceASC',
    label: 'ordenation.price.ascending',
  },
  {
    value: 'OrderByNameASC',
    label: 'ordenation.name.ascending',
  },
  {
    value: 'OrderByNameDESC',
    label: 'ordenation.name.descending',
  },
]

const OrderBy = ({ orderBy, getLinkProps, intl }) => {
  const sortingOptions = useMemo(() => {
    return SORT_OPTIONS.map(({ value, label }) => {
      return {
        value: value,
        label: intl.formatMessage({ id: label }),
      }
    })
  }, [intl])

  return (
    <div className={searchResult.orderBy}>
      <SelectionListOrderBy
        orderBy={orderBy}
        getLinkProps={getLinkProps}
        options={sortingOptions}
      />
    </div>
  )
}

OrderBy.propTypes = {
  /** Which sorting option is selected. */
  orderBy: PropTypes.string,
  /** Returns the link props. */
  getLinkProps: PropTypes.func,
  /** Intl instance. */
  intl: intlShape,
}

export default injectIntl(OrderBy)
