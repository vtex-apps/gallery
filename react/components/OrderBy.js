/* global __RUNTIME__ */
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { injectIntl, intlShape } from 'react-intl'
import { Dropdown } from 'vtex.styleguide'
import { withRuntimeContext } from 'render'

import SelectionListOrderBy from './SelectionListOrderBy'

export const SORT_OPTIONS = [
  {
    value: 'OrderByTopSaleDESC',
    label: 'ordenation.sales',
  },
  {
    value: 'OrderByReleaseDateDESC',
    label: 'ordenation.release.date',
  },
  {
    value: 'OrderByBestDiscountDESC',
    label: 'ordenation.discount',
  },
  {
    value: 'OrderByPriceDESC',
    label: 'ordenation.price.descending',
  },
  {
    value: 'OrderByPriceASC',
    label: 'ordenation.price.ascending',
  },
  {
    value: 'OrderByNameASC',
    label: 'ordenation.name.ascending',
  },
  {
    value: 'OrderByNameDESC',
    label: 'ordenation.name.descending',
  },
]

class OrderBy extends Component {
  static propTypes = {
    /** Which sorting option is selected. */
    orderBy: PropTypes.string.isRequired,
    /** Returns the link props. */
    getLinkProps: PropTypes.func,
    /** Intl instance. */
    intl: intlShape,
    /** Render Runtime context */
    runtime: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  }

  get sortingOptions() {
    return SORT_OPTIONS.map(opt => {
      return {
        value: opt.value,
        label: this.props.intl.formatMessage({ id: opt.label }),
      }
    })
  }

  render() {
    const { orderBy, getLinkProps, runtime } = this.props

    if (__RUNTIME__.hints.mobile) {
      return (
        <SelectionListOrderBy
          orderBy={orderBy}
          getLinkProps={getLinkProps}
          options={this.sortingOptions}
        />
      )
    }

    return (
      <Dropdown
        size="large"
        options={this.sortingOptions}
        value={orderBy}
        onChange={(_, ordenation) => {
          const pagesArgs = getLinkProps({ ordenation })
          runtime.navigate({
            page: pagesArgs.page,
            params: pagesArgs.params,
            query: pagesArgs.queryString,
            fallbackToWindowLocation: false,
          })
        }}
      />
    )
  }
}

export default withRuntimeContext(injectIntl(OrderBy))
