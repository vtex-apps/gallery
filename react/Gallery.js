import React, { useMemo, useContext } from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { pluck, splitEvery } from 'ramda'

import { useDevice } from 'vtex.device-detector'
import { ProductListContext } from 'vtex.product-list-context'
import { useResponsiveValue } from 'vtex.responsive-values'
import { useCssHandles } from 'vtex.css-handles'
import { useRuntime } from 'vtex.render-runtime'

import { LAYOUT_MODE } from './components/LayoutModeSwitcher'
import { productShape } from './constants/propTypes'
import withResizeDetector from './components/withResizeDetector'
import GalleryRow from './components/GalleryRow'
import ProductListEventCaller from './utils/ProductListEventCaller'
import SettingsContext from './components/SettingsContext'

import { Spinner } from 'vtex.styleguide'

/** Layout with one column */
const ONE_COLUMN_LAYOUT = 1

const LAZY_RENDER_THRESHOLD = 2

const CSS_HANDLES = ['gallery']

const { ProductListProvider } = ProductListContext
/**
 * Canonical gallery that displays a list of given products.
 */
const Gallery = ({
  products = [],
  mobileLayoutMode = 'normal',
  maxItemsPerRow = {
    desktop: 5,
    tablet: 3,
    phone: 2,
  },
  minItemWidth = 240,
  width,
  summary,
  showingFacets,
  lazyItemsRemaining,
}) => {
  const { isMobile } = useDevice()
  const { trackingId = 'Search result' } = useContext(SettingsContext) || {}
  const handles = useCssHandles(CSS_HANDLES)
  const { getSettings } = useRuntime()
  const responsiveMaxItemsPerRow = useResponsiveValue(maxItemsPerRow)

  const layoutMode = isMobile ? mobileLayoutMode : 'normal'

  const getItemsPerRow = () => {
    const maxItemsPerMinWidth = Math.floor(width / minItemWidth)

    if (isMobile) {
      if (layoutMode === 'normal') {
        return ONE_COLUMN_LAYOUT
      }

      const maxItemsOnMobile =
        maxItemsPerMinWidth >= 2 ? maxItemsPerMinWidth : 2

      return responsiveMaxItemsPerRow <= maxItemsOnMobile
        ? responsiveMaxItemsPerRow
        : maxItemsOnMobile
    }

    return responsiveMaxItemsPerRow <= maxItemsPerMinWidth
      ? responsiveMaxItemsPerRow
      : maxItemsPerMinWidth
  }

  const itemsPerRow = getItemsPerRow() || responsiveMaxItemsPerRow

  const rows = useMemo(() => splitEvery(itemsPerRow, products), [
    itemsPerRow,
    products,
  ])

  const galleryClasses = classNames(
    handles.gallery,
    'flex flex-row flex-wrap items-stretch bn ph1 na4',
    {
      'justify-center': !showingFacets,
      'pl9-l': showingFacets,
    }
  )

  const isLazyRenderEnabled = getSettings('vtex.store')
    ?.enableSearchRenderingOptimization

  return (
    <ProductListProvider listName={trackingId}>
      <div className={galleryClasses}>
        {rows.map((rowProducts, index) => (
          <GalleryRow
            key={index.toString()}
            widthAvailable={width != null}
            products={rowProducts}
            lazyRender={isLazyRenderEnabled && index >= LAZY_RENDER_THRESHOLD}
            summary={summary}
            displayMode={layoutMode}
            rowIndex={index}
            itemsPerRow={itemsPerRow}
            preloadFirstImage={index === 0}
          />
        ))}
        {typeof lazyItemsRemaining === 'number' && lazyItemsRemaining > 0 && (
          <div
            style={{
              width: '100%',
              // Approximate number, just to add scroll leeway
              height: 300 * Math.ceil(lazyItemsRemaining / itemsPerRow),
            }}
            className="flex justify-center pt10"
          >
            <Spinner />
          </div>
        )}
      </div>
      <ProductListEventCaller />
    </ProductListProvider>
  )
}

Gallery.propTypes = {
  /** Container width */
  width: PropTypes.number,
  /** Products to be displayed. */
  products: PropTypes.arrayOf(productShape),
  /** ProductSummary props. */
  summary: PropTypes.any,
  /** Max Items per Row */
  maxItemsPerRow: PropTypes.number,
  /** Layout mode of the gallery in mobile view */
  mobileLayoutMode: PropTypes.oneOf(pluck('value', LAYOUT_MODE)),
  /** Min Item Width. */
  minItemWidth: PropTypes.number,
  showingFacets: PropTypes.bool,
  trackingId: PropTypes.string,
  lazyItemsRemaining: PropTypes.number,
}

export default withResizeDetector(Gallery)
