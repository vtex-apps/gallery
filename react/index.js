import React, { Component } from 'react'

import SearchResultContainer from './components/SearchResultContainer'
import { SORT_OPTIONS } from './OrderBy'
import LocalQuery from './components/LocalQuery'
import { LAYOUT_MODE } from './components/LayoutModeSwitcher'
import QueryContext from './components/QueryContext'
import SettingsContext from './components/SettingsContext'

const PAGINATION_TYPES = ['show-more', 'infinite-scroll']
const DEFAULT_MAX_ITEMS_PER_PAGE = 10

/**
 * Search Result Query Loader Component.
 * This Component queries the search if the search-result doesn't receive it already
 */
export default class SearchResultQueryLoader extends Component {
  static defaultProps = {
    orderBy: SORT_OPTIONS[0].value,
    showCategoryPanel: false,
    quantityOfItemsPerRow: 4,
  }

  static uiSchema = {
    maxItemsPerLine: {
      'ui:widget': 'radio',
      'ui:options': {
        inline: true,
      },
    },
  }

  render() {
    const {
      querySchema,
      hiddenFacets,
      showFacetsQuantity,
      pagination,
      mobileLayout,
    } = this.props

    const settings = {
      hiddenFacets,
      showFacetsQuantity,
      pagination,
      mobileLayout,
    }

    return !this.props.searchQuery ||
      (querySchema && querySchema.enableCustomQuery) ? (
      <LocalQuery
        {...this.props}
        {...querySchema}
        render={props => (
          <QueryContext.Provider value={props.searchQuery.variables}>
            <SettingsContext.Provider value={settings}>
              <SearchResultContainer {...props} />
            </SettingsContext.Provider>
          </QueryContext.Provider>
        )}
      />
    ) : (
      <QueryContext.Provider value={this.props.searchQuery.variables}>
        <SettingsContext.Provider value={settings}>
          <SearchResultContainer {...this.props} />
        </SettingsContext.Provider>
      </QueryContext.Provider>
    )
  }
}

SearchResultQueryLoader.getSchema = props => {
  const querySchema = !props.searchQuery
    ? {
        querySchema: {
          title: 'admin/editor.search-result.query',
          description: 'admin/editor.search-result.query.description',
          type: 'object',
          properties: {
            maxItemsPerPage: {
              title: 'admin/editor.search-result.query.maxItemsPerPage',
              type: 'number',
              default: DEFAULT_MAX_ITEMS_PER_PAGE,
            },
            queryField: {
              title: 'Query',
              type: 'string',
            },
            mapField: {
              title: 'Map',
              type: 'string',
            },
            orderByField: {
              title: 'Order by field',
              type: 'string',
              default: SORT_OPTIONS[1].value,
              enum: SORT_OPTIONS.map(opt => opt.value),
              enumNames: SORT_OPTIONS.map(opt => opt.label),
            },
            hideUnavailableItems: {
              title: 'admin/editor.search-result.query.hideUnavailableItems',
              type: 'boolean',
              default: false,
            },
          },
        },
      }
    : {}

  return {
    title: 'admin/editor.search-result.title',
    description: 'admin/editor.search-result.description',
    type: 'object',
    properties: {
      ...querySchema,
      mobileLayout: {
        title: 'admin/editor.search-result.mobileLayout',
        type: 'object',
        isLayout: true,
        properties: {
          mode1: {
            title: 'admin/editor.search-result.mobileLayout.mode1',
            type: 'string',
            default: LAYOUT_MODE[0].value,
            enum: LAYOUT_MODE.map(opt => opt.value),
            enumNames: LAYOUT_MODE.map(opt => opt.label),
            isLayout: true,
          },
          mode2: {
            title: 'admin/editor.search-result.mobileLayout.mode2',
            type: 'string',
            default: LAYOUT_MODE[1].value,
            enum: LAYOUT_MODE.map(opt => opt.value),
            enumNames: LAYOUT_MODE.map(opt => opt.label),
            isLayout: true,
          },
        },
      },
      hiddenFacets: {
        title: 'admin/editor.search-result.hiddenFacets',
        type: 'object',
        isLayout: true,
        properties: {
          brands: {
            title: 'admin/editor.search-result.hiddenFacets.brands',
            type: 'boolean',
            isLayout: true,
          },
          categories: {
            title: 'admin/editor.search-result.hiddenFacets.categories',
            type: 'boolean',
            isLayout: true,
          },
          priceRange: {
            title: 'admin/editor.search-result.hiddenFacets.priceRange',
            type: 'boolean',
            isLayout: true,
          },
          specificationFilters: {
            title:
              'admin/editor.search-result.hiddenFacets.specificationFilters',
            type: 'object',
            isLayout: true,
            properties: {
              hideAll: {
                title:
                  'admin/editor.search-result.hiddenFacets.specificationFilters.hideAll',
                type: 'boolean',
                isLayout: true,
              },
              hiddenFilters: {
                type: 'array',
                isLayout: true,
                items: {
                  title:
                    'admin/editor.search-result.hiddenFacets.specificationFilters.hiddenFilter',
                  type: 'object',
                  isLayout: true,
                  properties: {
                    name: {
                      title:
                        'admin/editor.search-result.hiddenFacets.specificationFilters.hiddenFilter.name',
                      type: 'string',
                      isLayout: true,
                    },
                  },
                },
              },
            },
          },
        },
      },
      pagination: {
        type: 'string',
        title: 'admin/editor.search-result.pagination.title',
        default: 'infinity-scroll',
        enum: PAGINATION_TYPES,
        enumNames: [
          'admin/editor.search-result.pagination.show-more',
          'admin/editor.search-result.pagination.infinite-scroll',
        ],
      },
      showCategoryPanel: {
        title: 'editor.search-result.show-category-panel.title',
        type: 'boolean',
        default: false,
        isLayout: true,
      },
      quantityOfItemsPerRow: {
        type: 'number',
        title: 'editor.search-result.quantity-of-items-per-row.title',
        enum: [2, 4],
        default: 4,
        widget: {
          'ui:widget': 'radio',
          'ui:options': {
            inline: true,
          },
        },
        isLayout: true,
      },
      showFacetQuantity: {
        type: 'boolean',
        title: 'admin/editor.search-result.facet-quantity',
        default: false,
      },
    },
  }
}
