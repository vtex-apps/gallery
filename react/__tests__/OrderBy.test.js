/* eslint-env jest */
import React from 'react'
import { render, fireEvent } from '@vtex/test-tools/react'

import OrderBy from '../OrderBy'
import { setMobileState } from 'vtex.render-runtime'

describe('<OrderBy />', () => {
  const renderComponent = customProps => mobile => {
    setMobileState(mobile)

    const props = {
      orderBy: 'OrderByTopSaleDESC',
      ...customProps,
    }

    return render(<OrderBy {...props} />)
  }

  it('should shown dropdown box on mobile mode', () => {
    const { container } = renderComponent()(true)

    let dropdownContainer = container.querySelector('.dropdownSort > .dn')
    const button = container.querySelector('button')

    // Expect Dropdown container be rendered with display none
    expect(dropdownContainer).toBeDefined()
    expect(dropdownContainer).not.toBeNull()

    fireEvent.click(button)
    dropdownContainer = container.querySelector('.dropdownSort > .dn')

    // Expect Dropdown container be rendered without display none
    expect(dropdownContainer).toBeNull()
  })

  it('should match snapshot in web mod', () => {
    const { asFragment } = renderComponent()(false)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot in mobile', () => {
    const { asFragment } = renderComponent()(true)

    expect(asFragment()).toMatchSnapshot()
  })

  afterEach(() => {
    setMobileState(false)
  })
})
