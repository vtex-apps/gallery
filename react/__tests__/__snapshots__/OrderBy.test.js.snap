// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderBy /> should match snapshot in mobile 1`] = `
<DocumentFragment>
  <div
    class="orderBy"
  >
    <div
      class="dropdownSort relative pt1 justify-center w-100 w-auto-ns center"
    >
      <button
        class="ph3 pv5 mv0 pointer flex justify-center items-center bg-base c-on-base t-action--small bt br bl bb-0 br2 br--top bw1 w-100 b--transparent pl1"
      >
        <span
          class="filterPopupTitle c-on-base t-action--small ml-auto-ns"
        >
          <span
            class="c-muted-2"
          >
            Sort By
          </span>
           Sales
        </span>
        <span
          class="filterPopupArrowIcon ph5 pt1"
        >
          <div>
            IconCaret
          </div>
        </span>
      </button>
      <div
        class="z-1 absolute bg-base shadow-5 f5 w-100 b--muted-4 br2 ba bw1 br--bottom dn"
      >
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Sort By
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Sales
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Release Date
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Discount
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Price: High to Low
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Price: Low to High
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Name, ascending
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Name, descending
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<OrderBy /> should match snapshot in web mod 1`] = `
<DocumentFragment>
  <div
    class="orderBy"
  >
    <div
      class="dropdownSort relative pt1 justify-center w-100 w-auto-ns center"
    >
      <button
        class="ph3 pv5 mv0 pointer flex justify-center items-center bg-base c-on-base t-action--small bt br bl bb-0 br2 br--top bw1 w-100 b--transparent pl1"
      >
        <span
          class="filterPopupTitle c-on-base t-action--small ml-auto-ns"
        >
          <span
            class="c-muted-2"
          >
            Sort By
          </span>
           Sales
        </span>
        <span
          class="filterPopupArrowIcon ph5 pt1"
        >
          <div>
            IconCaret
          </div>
        </span>
      </button>
      <div
        class="z-1 absolute bg-base shadow-5 f5 w-100 b--muted-4 br2 ba bw1 br--bottom dn"
      >
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Sort By
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Sales
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Release Date
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Discount
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Price: High to Low
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Price: Low to High
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Name, ascending
        </button>
        <button
          class="bg-base c-on-base f5 ml-auto db no-underline pointer tl bn pv4 ph5 hover-bg-muted-4 w-100"
        >
          Name, descending
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
